File: Project Templates/Mac/Application Extension/Audio Unit Extension.xctemplate/AudioUnitViewController.m
//
//  ___FILENAME___
//  ___PACKAGENAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

#import "___FILEBASENAME___.h"
#import "MyAudioUnit.h"

@interface ___FILEBASENAME___ ()

@end

@implementation ___FILEBASENAME___ {
    AUAudioUnit *audioUnit;
}

- (void) viewDidLoad {
    [super viewDidLoad];
    
    if (!audioUnit) {
        return;
    }
    
    // Get the parameter tree and add observers for any parameters that the UI needs to keep in sync with the AudioUnit
}

- (AUAudioUnit *)createAudioUnitWithComponentDescription:(AudioComponentDescription)desc error:(NSError **)error {
    audioUnit = [[MyAudioUnit alloc] initWithComponentDescription:desc error:error];
    
    return audioUnit;
}

@end

File: Project Templates/Mac/Application Extension/Audio Unit Extension.xctemplate/AudioUnitViewController.swift
//
//  ___FILENAME___
//  ___PACKAGENAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

import CoreAudioKit

public class ___FILEBASENAME___: AUViewController, AUAudioUnitFactory {
    var audioUnit: AUAudioUnit?
    
    public override func viewDidLoad() {
        super.viewDidLoad()
        
        if audioUnit == nil {
            return
        }
        
        // Get the parameter tree and add observers for any parameters that the UI needs to keep in sync with the AudioUnit
    }
    
    public func createAudioUnitWithComponentDescription(componentDescription: AudioComponentDescription) throws -> AUAudioUnit {
        audioUnit = try MyAudioUnit(componentDescription: componentDescription, options: [])
        
        return audioUnit!
    }
    
}
